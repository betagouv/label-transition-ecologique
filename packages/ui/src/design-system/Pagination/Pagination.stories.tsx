import {Meta, StoryObj} from '@storybook/react';
import {Pagination} from '.';

const meta: Meta<typeof Pagination> = {
  component: Pagination,
  args: {
    selectedPage: 1,
    nbOfPages: 5,
  },
};

export default meta;

type Story = StoryObj<typeof Pagination>;

export const Default: Story = {};

export const ManyPages: Story = {
  args: {nbOfPages: 10},
};
