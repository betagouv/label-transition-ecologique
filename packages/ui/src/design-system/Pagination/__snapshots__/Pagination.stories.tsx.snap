// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`design-system/Pagination Default smoke-test 1`] = `
<nav class="flex w-fit gap-6">
  <button title="Page précédente"
          disabled
          class="w-fit flex items-center border-solid group gap-1 cursor-not-allowed rounded-lg border font-bold !text-primary-3 !bg-primary-0 !border-primary-4 text-xs py-2 px-5"
  >
    <span class="ri-arrow-left-s-line font-normal text-center h-4 w-4 text-[1rem] leading-[1rem] fill-primary-3">
    </span>
    Précédent
  </button>
  <div class="flex gap-3">
    <button aria-current="page"
            title="Page 1"
            class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 !p-2 !w-9 justify-center"
    >
      1
    </button>
    <button aria-current="page"
            title="Page 2"
            class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 !p-2 !w-9 justify-center"
    >
      2
    </button>
    <button aria-current="page"
            title="Page 3"
            class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 !p-2 !w-9 justify-center"
    >
      3
    </button>
    <button aria-current="page"
            title="Page 4"
            class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 !p-2 !w-9 justify-center"
    >
      4
    </button>
    <button aria-current="page"
            title="Page 5"
            class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 !p-2 !w-9 justify-center"
    >
      5
    </button>
  </div>
  <button title="Page suivante"
          class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 flex-row-reverse"
  >
    <span class="ri-arrow-right-s-line font-normal text-center h-4 w-4 text-[1rem] leading-[1rem] fill-primary-7 group-hover:fill-primary-8">
    </span>
    Suivant
  </button>
</nav>
`;

exports[`design-system/Pagination ManyPages smoke-test 1`] = `
<nav class="flex w-fit gap-6">
  <button title="Page précédente"
          disabled
          class="w-fit flex items-center border-solid group gap-1 cursor-not-allowed rounded-lg border font-bold !text-primary-3 !bg-primary-0 !border-primary-4 text-xs py-2 px-5"
  >
    <span class="ri-arrow-left-s-line font-normal text-center h-4 w-4 text-[1rem] leading-[1rem] fill-primary-3">
    </span>
    Précédent
  </button>
  <div class="flex gap-3">
    <button aria-current="page"
            title="Page 1"
            class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-white hover:!text-white bg-primary-7 hover:!bg-primary-8 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 !p-2 !w-9 justify-center"
    >
      1
    </button>
    <button aria-current="page"
            title="Page 2"
            class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 !p-2 !w-9 justify-center"
    >
      2
    </button>
    <button aria-current="page"
            title="Page 3"
            class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 !p-2 !w-9 justify-center"
    >
      3
    </button>
    <button aria-current="page"
            title="Page undefined"
            disabled
            class="w-fit flex items-center border-solid group gap-1 cursor-not-allowed rounded-lg border font-bold !text-primary-3 !bg-primary-0 !border-primary-4 text-xs py-2 px-5 !p-2 !w-9 justify-center"
    >
      ...
    </button>
    <button aria-current="page"
            title="Page 8"
            class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 !p-2 !w-9 justify-center"
    >
      8
    </button>
    <button aria-current="page"
            title="Page 9"
            class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 !p-2 !w-9 justify-center"
    >
      9
    </button>
    <button aria-current="page"
            title="Page 10"
            class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 !p-2 !w-9 justify-center"
    >
      10
    </button>
  </div>
  <button title="Page suivante"
          class="w-fit flex items-center border-solid group gap-1 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-xs py-2 px-5 flex-row-reverse"
  >
    <span class="ri-arrow-right-s-line font-normal text-center h-4 w-4 text-[1rem] leading-[1rem] fill-primary-7 group-hover:fill-primary-8">
    </span>
    Suivant
  </button>
</nav>
`;
